Extractor/FrontendExtractor.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Bazinga\Bundle\JsTranslationBundle\Extractor;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Translation\Extractor\AbstractFileExtractor;
+use Symfony\Component\Translation\MessageCatalogue;
+use Symfony\Component\Translation\Extractor\ExtractorInterface;
+
+final class FrontendExtractor extends AbstractFileExtractor implements ExtractorInterface
+{
+    const FUNCTION_STRING_ARGUMENT_REGEX = '\s?[\'"]([^"\'),]+)[\'"]\s?';
+    const REGEX_DELIMITER = '/';
+
+    private $prefix = '';
+
+    private $filesystem;
+
+    private $configuration;
+
+    public function __construct(Filesystem $filesystem, array $configuration = []) {
+        $this->filesystem = $filesystem;
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function extract($resource, MessageCatalogue $catalogue)
+    {
+        $files = $this->extractFiles($resource);
+
+        foreach ($files as $file) {
+            foreach ($this->configuration as $configuration)
+            {
+              $pathInfo = pathinfo($file);
+              if (in_array($pathInfo['extension'], $configuration['extensions'], true))
+              {
+                $this->parseMessagesFromContent(file_get_contents($file), $catalogue, $configuration['sequence']);
+              }
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setPrefix($prefix)
+    {
+        $this->prefix = $prefix;
+    }
+
+    private function parseMessagesFromContent($fileContent, MessageCatalogue $catalogue, $sequence)
+    {
+        $messages = $this->getMessagesForSequence($fileContent, $sequence);
+
+        foreach ($messages as $message) {
+            $catalogue->set($message, $this->prefix.$message);
```

Comment:
Setting them always in the `messages` domain is wrong. The best solution would be to support extracting the domain properly (but this is hard). But the basic solution is at least to respect the default domain being configured (which may be changed to something else than `messages` which is very handy when your frontend needs less than 1% of all your translation messages)

Reply:
Then should we specify the domain in the constructor? (as the TranslationDumper does)
You said we can use the default domain, do you know the way to retrieve it frome this part of the code?

Reply:
The default domain should indeed be injected in the constructor.